src/blk/zoned/HMSMRDevice.cc

```diff
@@ -412,6 +412,14 @@ void HMSMRDevice::_detect_vdo()
   return;
 }
 
+// Will reset the write pointer of all zones from start to end(both inclusive)
+bool HMSMRDevice::reset_zones(uint64_t zone_num_range_start, uint64_t zone_num_range_end) {
+  ceph_assert(is_smr());
+  uint64_t len = (zone_num_range_end + 1 - zone_num_range_start)  * zone_size;
+  ceph_assert(len > 0);
+  return !zbd_reset_zones(zbd_file_descriptor, zone_num_range_start * zone_size, len);
```

Comment:
In case of failure it would be useful to print the error message.

Reply:
Added